pkg/util/validation/validation.go

```diff
@@ -55,18 +56,37 @@ func validateResourceSelectorsIfPreemptionEnabled(spec policyv1alpha1.Propagatio
 	var allErrs field.ErrorList
 	for index, resourceSelector := range spec.ResourceSelectors {
 		if len(resourceSelector.Name) == 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Index(index).Child("name"), resourceSelector.Name, "name can not be empty if preemption is Always, the empty name may cause unexpected resources preemption"))
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(index).Child("name"), resourceSelector.Name, "name cannot be empty if preemption is Always, the empty name may cause unexpected resources preemption"))
 		}
 	}
 	return allErrs
 }
 
+// validateSuspension validates no conflicts between dispatching and dispatchingOnClusters.
+func validateSuspension(suspension *policyv1alpha1.Suspension, fldPath *field.Path) field.ErrorList {
+	if suspension == nil {
+		return nil
+	}
+
+	if !ptr.Deref(suspension.Dispatching, false) {
+		return nil
+	}
+
+	if suspension.DispatchingOnClusters == nil || len(suspension.DispatchingOnClusters.ClusterNames) == 0 {
+		return nil
+	}
+
+	return field.ErrorList{
+		field.Invalid(fldPath.Child("suspension"), suspension, "suspension dispatching cannot co-exist with dispatchingOnClusters.clusterNames"),
+	}
```

Comment:
I would prefer this straightforward style:
```suggestion
	if (suspension.Dispatching != nil && *suspension.Dispatching == true) &&
		(suspension.DispatchingOnClusters != nil && len(suspension.DispatchingOnClusters.ClusterNames) > 0) {

		return field.ErrorList{
			field.Invalid(fldPath.Child("suspension"), suspension, "suspension dispatching cannot co-exist with dispatchingOnClusters.clusterNames"),
		}
	}

	return nil
```

Reply:
the old version was easier to digest for me ... ideally 1 comment per `if` explaining what is going on
suggested feels a overwhelming

Reply:
I'm sorry for not explaining why I'm suggesting that in detail.

Firstly the old version is great, it looks simple. But given that we need a single check on whether the two fields co-exist, I tend to a more cohesive approach to make it easier to add more checks in the future.